### ISSUE TITLE
Better error logging for Invalid Struct

### ISSUE THREAD

**Issue by dblythy** [parse-community/Parse-Swift#237]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-Swift/security/policy).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-Swift/issues?q=is%3Aissue).

### Current Limitation

At the moment, if you define a struct which doesn't match your server payload, you get no meaningful errors that help you isolate the problem.

### Feature / Enhancement Description

Add a `.invalidStruct` error with information as to where the struct doesn't match the server payload

### Example Use Case

-

### Alternatives / Workarounds

-

### 3rd Party References

-

**Comment by cbaker6** [comment]
@dblythy thanks for opening. Can you provide and example of an “invalid struct”? I’ve never created or seen an invalid struct before while using the SDK so I’m interested to see what it looks like. Also, can you post the error or information the SDK currently provides before your PR.

**Comment by dblythy** [comment]
Yes, so it's if your struct doesn't match up with your Parse data that is returned in a query. This happens if the data is already stored using an alternate method, such as the JS SDK, and you're building the `struct` to match the existing data (such as invalid type or not set to optional). At the moment the error is:

```
"Error decoding parse-server response: Optional(<NSHTTPURLResponse: 0x6000019d1100> { URL: http://localhost:1337/1/classes/GameScore } {
```

This changes the error to:

```
"Invalid struct: No value associated with key CodingKeys(stringValue: \"score\", intValue: nil) (\"score\")."
```

**Comment by cbaker6** [comment]
It seems that you can add your comment https://github.com/parse-community/Parse-Swift/issues/237#issuecomment-922748352 in the issue description for "Example Use Case" and "Alternative Workarounds"
